cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/DefaultApplicationUpdateContext.java

```diff
@@ -0,0 +1,164 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app;
+
+import com.google.common.base.Predicate;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+import io.cdap.cdap.api.Config;
+import io.cdap.cdap.api.app.ApplicationConfigUpdateAction;
+import io.cdap.cdap.api.app.ApplicationUpdateContext;
+import io.cdap.cdap.api.artifact.ArtifactId;
+import io.cdap.cdap.api.artifact.ArtifactScope;
+import io.cdap.cdap.api.artifact.ArtifactVersion;
+import io.cdap.cdap.api.artifact.ArtifactVersionRange;
+import io.cdap.cdap.api.plugin.PluginClass;
+import io.cdap.cdap.common.id.Id.Artifact;
+import io.cdap.cdap.common.id.Id.Namespace;
+import io.cdap.cdap.common.io.CaseInsensitiveEnumTypeAdapterFactory;
+import io.cdap.cdap.internal.app.runtime.artifact.ArtifactDescriptor;
+import io.cdap.cdap.internal.app.runtime.artifact.ArtifactRepository;
+import io.cdap.cdap.proto.artifact.ArtifactSortOrder;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.id.NamespaceId;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Default implementation of {@link ApplicationUpdateContext}.
+ *
+ * Used during update of an Application config via provided helper methods used during an update action like
+ * upgrade/downgrade.
+ */
+public class DefaultApplicationUpdateContext implements ApplicationUpdateContext {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultApplicationUpdateContext.class);
+  private static final Gson GSON = new GsonBuilder().
+      registerTypeAdapterFactory(new CaseInsensitiveEnumTypeAdapterFactory()).create();
+
+  private final ArtifactId applicationArtifactId;
+  private String configString;
+  private List<ApplicationConfigUpdateAction> updateActions;
+  private final ArtifactRepository artifactRepository;
+  private final ApplicationId applicationId;
+  private final NamespaceId namespaceId;
+
+  public DefaultApplicationUpdateContext(NamespaceId namespaceId, ApplicationId applicationId,
+                                         ArtifactId applicationArtifactId, ArtifactRepository artifactRepository,
+                                         String configString, List<ApplicationConfigUpdateAction> updateActions) {
+    this.namespaceId = namespaceId;
+    this.applicationId = applicationId;
+    this.artifactRepository = artifactRepository;
+    this.applicationArtifactId = applicationArtifactId;
+    this.configString = configString;
+    this.updateActions = updateActions;
+  }
+
+  @Override
+  public List<ApplicationConfigUpdateAction> getUpdateActions() {
+    return updateActions;
+  }
+
+  @Override
+  public <C extends Config> C getConfig(Type configType) {
+    C appConfig;
```

Comment:
No need to have this local file. You can just return directly at the place where you construct the new instance.